aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,211 @@
-.. include:: /../common/stub-topic.txt
+Model Validation 
+================
+By `Rachel Appel <http://github.com/rachelappel>`_
 
-|stub-icon| Model Validation
-============================
+In this article:
 
-.. include:: /../common/stub-notice.txt
+.. contents:: Sections
+  :local:
+  :depth: 1
 
-.. _issue: https://github.com/aspnet/Docs/issues/116
+Introduction to model validation
+--------------------------------
+
+Before an app stores data in a database, the app must validate the data. Data must be checked for potential security threats, verified that it is appropriately formatted by type and size, and it must conform to your rules. Validation is necessary although it can be redundant and tedious to implement. In MVC, validation happens on both the client and server. 
+
+Fortunately, .NET has abstracted validation into validation attributes. These attributes contain validation code, thereby reducing the amount of code you must write. 
+
+Validation Attributes
+---------------------
+
+Validation attributes are a way to configure model validation so it's similar conceptually to validation on fields in tables. This includes constraints such as assigning data types or required fields. Other types of validation include applying patterns to data to enforce business rules, such as a credit card, phone number, or email address. Validation attributes make enforcing these requirements much simpler and easier to use.
+
+Below is an annotated ``Movie`` model from an app that stores information about movies and TV shows. Most of the properties are required and several string properties have length requirements. Additionally, there is a numeric range restriction in place for the ``Price`` property from 0 to $999.99, along with a custom annotation.
+
+.. literalinclude:: validation/sample/Movie.cs
+   :language: c#
+   :lines: 7-34
+   :dedent: 4
+
+Simply reading through the model reveals the rules about data for this app, making it easier to maintain the code. Below are several popular built-in validation attributes:
+
+- ``[CreditCard]``: Validates the property as a credit card type.
+- ``[Compare]``: Compares two properties in a model. It works with various types, such as int, string, or date. 
+- ``[EmailAddress]``: Validates the property with an email format.
+- ``[Phone]``: Validates the property as a telephone format.
+- ``[Range]``: Sets the minimum and maximum values for multiple types of properties.
```

Comment:
"Validates the property value falls within the given range." Up to you whether "Works with multiple property types." is needed.